### ISSUE TITLE
when net trainning many epochs, batch outputs is the same

### ISSUE THREAD

**Issue by Cugtyt** [ericsun99/Shufflenet-v2-Pytorch-#3]
I use width_mult=0.5, input batch shape is (9, 15, 224, 224), each input element is (15, 224, 224), 5 imgs together, and output is 63 classes.

My train strategy is like this:
``` python
for i in range(epoch):
    train, label = get_data()
    train the net 20 times # because get train and label is time expensive
    # the criterion is nn.CrossEntropyLoss()
    # and the optimizer is optim.Adam(model.parameters())
```

but I print the predict and true label, I found that when trainning enough epochs,  the output of one epoch is the same, this is what I got:

```
[ 6 13 20 27 34 41 48 55 62]  <<<< true label >>>>
tensor([ 4, 26, 39,  5, 37, 17, 26, 60,  9], device='cuda:0')   <<<< predcition >>>>
1938: loss:  2.8527412176132203 acc: 0.3277777777777778
[ 0  7 14 21 28 35 42 49 56]
tensor([59, 40, 13, 45,  5, 60, 20, 52, 31], device='cuda:0')
1939: loss:  2.920798599720001 acc: 0.31111111111111117
[ 1  8 15 22 29 36 43 50 57]
tensor([42, 56, 55, 40,  5,  5, 60, 20, 62], device='cuda:0')
1940: loss:  2.573692339658737 acc: 0.3777777777777778
[ 2  9 16 23 30 37 44 51 58]
tensor([18, 18,  7,  0,  1, 35,  1, 50,  8], device='cuda:0')
1941: loss:  2.1535364985466003 acc: 0.4833333333333333
[ 3 10 17 24 31 38 45 52 59]
tensor([16, 23, 17, 42, 38, 57, 15, 50, 13], device='cuda:0')
1942: loss:  2.651383513212204 acc: 0.41111111111111115
[ 4 11 18 25 32 39 46 53 60]
tensor([ 7,  7,  9,  1, 37, 23, 24, 29, 44], device='cuda:0')
1943: loss:  2.331027537584305 acc: 0.4222222222222222
[ 5 12 19 26 33 40 47 54 61]
tensor([51, 37, 15,  9, 38, 10, 51, 59, 17], device='cuda:0')
1944: loss:  2.423956948518753 acc: 0.4166666666666667
[ 6 13 20 27 34 41 48 55 62]
tensor([52, 17,  3, 32, 11,  1, 52, 56, 17], device='cuda:0')
1945: loss:  2.5490114033222198 acc: 0.4555555555555556
[ 0  7 14 21 28 35 42 49 56]
tensor([56, 19, 59, 19, 43, 17, 32, 26, 55], device='cuda:0')
1946: loss:  2.4000859677791597 acc: 0.45555555555555555
[ 1  8 15 22 29 36 43 50 57]
tensor([13, 61, 17, 14, 14, 32, 24, 19,  8], device='cuda:0')
1947: loss:  3.444172370433807 acc: 0.2611111111111111
[ 2  9 16 23 30 37 44 51 58]
tensor([50, 50, 50, 50, 50, 50, 50, 50, 50], device='cuda:0')  <<<<<<<<< predict all the same
1948: loss:  4.35312852859497 acc: 0.04444444444444445
[ 3 10 17 24 31 38 45 52 59]
tensor([44, 44, 44, 44, 44, 44, 44, 44, 44], device='cuda:0')
1949: loss:  4.460417032241821 acc: 0.0
[ 4 11 18 25 32 39 46 53 60]
tensor([44, 44, 44, 44, 44, 44, 44, 44, 44], device='cuda:0')
1950: loss:  4.368063020706177 acc: 0.0
[ 5 12 19 26 33 40 47 54 61]
tensor([59, 59, 59, 59, 59, 59, 59, 59, 59], device='cuda:0')
1951: loss:  4.322405076026916 acc: 0.0
[ 6 13 20 27 34 41 48 55 62]
tensor([39, 39, 39, 39, 39, 39, 39, 39, 39], device='cuda:0')
1952: loss:  4.31459538936615 acc: 0.0
[ 0  7 14 21 28 35 42 49 56]
tensor([12, 12, 12, 12, 12, 12, 12, 12, 12], device='cuda:0')
1953: loss:  4.37600839138031 acc: 0.0
[ 1  8 15 22 29 36 43 50 57]
tensor([6, 6, 6, 6, 6, 6, 6, 6, 6], device='cuda:0')
1954: loss:  4.262833166122436 acc: 0.0
[ 2  9 16 23 30 37 44 51 58]
tensor([48, 48, 48, 48, 48, 48, 48, 48, 48], device='cuda:0')
1955: loss:  4.157500886917115 acc: 0.0
[ 3 10 17 24 31 38 45 52 59]
tensor([43, 43, 43, 43, 43, 43, 43, 43, 43], device='cuda:0')
1956: loss:  4.2394838571548465 acc: 0.04444444444444445
```

Why is that? I have no idea about it.

**Comment by mbaharan** [comment]
Hi, did you solve the problem? I have the same situation. If I train the network in python 2.7, it is going to work, however, in python3 the network is not going to converge, and the loss function is going to be same for the rest of epochs. Couldn't figure it out.

**Comment by lilingge** [comment]
> Hi, did you solve the problem? I have the same situation. If I train the network in python 2.7, it is going to work, however, in python3 the network is not going to converge, and the loss function is going to be same for the rest of epochs. Couldn't figure it out.

Hi, I have the same problem! the loss is going to be same. And I find the memory of GPU only cost a little when the bacthsize is 128, the image is 3 224 224.
